src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ReferenceTrackerHost.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Runtime.InteropServices
+{
+    internal static class ReferenceTrackerHost
+    {
+        [FixedAddressValueType]
+        internal static readonly unsafe IntPtr s_globalHostServices = (IntPtr)Unsafe.AsPointer(in HostServices.Vftbl);
+
+        // Called when an IReferenceTracker instance is found.
+        public static void SetReferenceTrackerHost(IntPtr trackerManager)
+        {
+            IReferenceTrackerManager.SetReferenceTrackerHost(trackerManager, s_globalHostServices);
```

Comment:
```suggestion
            IReferenceTrackerManager.SetReferenceTrackerHost(trackerManager, (IntPtr)Unsafe.AsPointer(in s_globalHostServices));
```
Is this missing an indirection?